tests/GeneratorTest.php

```diff
@@ -31,7 +34,33 @@ public function it_generates_open_api_spec_for_route()
 
         $this->assertNotNull($spec['paths']['/things']['post'] ?? null);
 
-        $this->assertEquals('postThingResource', $spec['paths']['/things']['post']['operationId'] ?? null);
+        $this->assertEquals('postThing', $spec['paths']['/things']['post']['operationId'] ?? null);
+    }
+
+    /**
+     * @test
+     *
+     * @define-route nestedRoutes
+     */
+    public function it_generates_distinct_operation_ids_with_nested_routes()
+    {
+        /** @var Generator $generator */
+        $generator = app(Generator::class);
+        $spec = json_decode($generator->generate(), true);
+
+        $this->assertEquals('postPet', Arr::get($spec, 'paths./pets.post.operationId'));
+        $this->assertEquals('postPetOwner', Arr::get($spec, 'paths./pets/{pet}/owner.post.operationId'));
+    }
+
+    protected function nestedRoutes($app)
+    {
+        /** @var Router $router */
+        $router = $app->make('router');
+        $router->middleware(['api'])->group(function (Router $router) {
+            $router->post('pets', [PetController::class, 'store']);
+
+            $router->post('pets/{pet}/owner', [PetController::class, 'storeOwner']);
+        });
```

Comment:
entity name is nu ook gefixt toch voor urls die lijken `/resoure/{id}`?
Wil je daar een test voor hebben?

Reply:
I saw that index and show methods would get the same operation ids now and I fixed it. Also included in the tests.